make.js

```diff
@@ -211,29 +209,23 @@ CLI.serverBuild = function(/** @type {{ node: string; task: string }} */ argv) {
 
     // Ensure we wrap build function after generator's changes to store only files that changes after the build 
     const buildTaskWrapped = util.syncGeneratedFilesWrapper(buildTask, genTaskPath, callGenTaskDuringBuild);
-    const nodeVersion = (argv.node && argv.node == "Node20") ? argv.node : "Default";
     const allTasksNode20 = allTasks.filter((taskName) => {
-        return taskName.endsWith("Node20");
+        return getNodeVersion(buildTasksPath, taskName) == 20;
```

Comment:
[buildTasksPath](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make.js#L44) is the path to _build/Tasks where the built tasks are placed, probably, you need [tasksPath](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make.js#L48) for the tasks sources and [genTaskPath](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make.js#L59) for the generated sources. the getNodeVersion function's check could be something like this:
![image](https://github.com/microsoft/azure-pipelines-tasks/assets/106314398/e5c80deb-72fc-46fb-a68d-96ce3d4d7412)

Reply:
Updated